### PR TITLE
[KYUUBI #2543] Make rowCountPerTask configurable

### LINKED ISSUE #2543: [Subtask] Kyuubi Spark TPC-DS Connector - Make inputPartitionSize configurable
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-kyuubi/issues) and found no similar issues.

### Describe the subtask

Currently `rowCountPerTask` is hardcoded as 1 million, which impacts the parallelism and final impacts the performance.

```
  private val rowCountPerTask: Int = 1000000

  private val rowCount: Long = new Scaling(scale).getRowCount(table)

  private val parallelism: Int =
    if (table.isSmall) 1
    else math.max(
      SparkSession.active.sparkContext.defaultParallelism,
      (rowCount / rowCountPerTask.toDouble).ceil.toInt)

```

After discussion, we need to introduce a configuration to control the parallelism based on scan partition size

### Parent issue

https://github.com/apache/incubator-kyuubi/issues/2538

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by wForget**
### _Why are the changes needed?_

close #2543

### _How was this patch tested?_
- [X] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by wForget** [comment]
cc @pan3793

File: extensions/spark/kyuubi-spark-connector-tpcds/src/main/scala/org/apache/kyuubi/spark/connector/tpcds/TPCDSTable.scala [hunk 1]
```diff
@@ -68,8 +68,13 @@ class TPCDSTable(tbl: String, scale: Int, options: CaseInsensitiveStringMap)
...
     Set(TableCapability.BATCH_READ).asJava
 
-  override def newScanBuilder(options: CaseInsensitiveStringMap): ScanBuilder = {
-    new TPCDSBatchScan(tpcdsTable, scale, schema)
+  override def newScanBuilder(scanOpts: CaseInsensitiveStringMap): ScanBuilder = {
+    val map = new util.HashMap[String, String](scanOpts)
+    options.asScala.foreach {
+      case (k, v) => map.putIfAbsent(k, v)
+    }
+    val newOpts = new CaseInsensitiveStringMap(map)
```
Comment (pan3793) [inline extensions/spark/kyuubi-spark-connector-tpcds/src/main/scala/org/apache/kyuubi/spark/connector/tpcds/TPCDSTable.scala:76]: It's good to consider both SparkConf and ReadOption, can we extract it such as `org.apache.iceberg.spark.SparkConfParser`?

Comment (wForget) [reply extensions/spark/kyuubi-spark-connector-tpcds/src/main/scala/org/apache/kyuubi/spark/connector/tpcds/TPCDSTable.scala:76]: Sounds good, I will try it later.

**Comment by pan3793** [comment]
I notice that trinodb/tpcds has a hardcode restriction for split, let's add some comments as well.
https://github.com/trinodb/tpcds/blob/8a02abbba864feedc2afd078c8153d66a95bb2d4/src/main/java/io/trino/tpcds/Parallel.java#L26-L36

**Comment by pan3793** [comment]
https://github.com/trinodb/tpcds/issues/3

**Comment by wForget** [comment]
https://github.com/apache/incubator-kyuubi/issues/2543#issuecomment-1132427303

**Comment by pan3793** [comment]
Thanks @wForget, don't forget to update the umbrella since we change the plan :)

**Comment by wForget** [comment]
> Thanks @wForget, don't forget to update the umbrella since we change the plan :)

Thank you for your reminder. I will update the issue title and umbrella.

**Pull request closed without merging**